### PR TITLE
Refactoring of validation rules to include custom filters

### PR DESCRIPTION
**Pull request by GemTay**
# Purpose

Adds custom filters to the activation key and viewer code fields to then perform simpler validation on the filtered input

## Approach

Creates custom filters for the activation key and viewer code, to strip out prefixes/whitespace/hyphens. By doing this filtering first, it allows us to have simpler validation on the input. For example, before we had to set the input length validation min=12 and max=23 as users input could include hyphens and spaces eg C - ABCD - 1234 - EFGH. With the new filters we can now be strict on the validation length set to 12. The filters also prevent having to filter the data in the handlers.

## Checklist

* [x] I have performed a self-review of my own code
* [ ] I have added relevant logging with appropriate levels to my code
* [ ] I have updated documentation (Confluence/GitHub wiki/tech debt doc) where relevant
* [x] I have added tests to prove my work
* [ ] I have added welsh translation tags and updated translation files
* [ ] I have run an accessibility tool on any pages I have made changes to and fixed any issues found
* [ ] The product team have tested these changes

### CODE REVIEW

File: service-front/app/src/Common/src/Filter/ActivationKeyFilter.php [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+use Laminas\Filter\AbstractFilter;
+
+class ActivationKeyFilter extends AbstractFilter
+{
+    /**
+     * @param string $activationKey
+     * @return string
+     */
+    public function filter($activationKey): string
```
Comment (cooperaj) [inline service-front/app/src/Common/src/Filter/ActivationKeyFilter.php:15]:
```suggestion
    public function filter(string $activationKey): string
```

Comment (GemTay) [reply service-front/app/src/Common/src/Filter/ActivationKeyFilter.php:15]: I can't define the input type unfortunately, theres an error :( `Declaration must be compatible with FilterInterface->filter(value)`

File: service-front/app/src/Actor/templates/actor/form-error-messages.html.twig [hunk 2]
```diff
@@ -29,6 +29,8 @@
 {# Add lpa #}
 {% trans %}Enter your activation key{% context %}error{% notes %}Add LPA{% endtrans %}
 {% trans %}Enter an activation key in the correct format{% context %}error{% notes %}Add LPA{% endtrans %}
+{% trans %}The activation key you entered is too long{% context %}error{% notes %}Add LPA{% endtrans %}
+{% trans %}The activation key you entered is too short{% context %}error{% notes %}Add LPA{% endtrans %}
```
Comment (GemTay) [inline service-front/app/src/Actor/templates/actor/form-error-messages.html.twig:33]: Waiting on welsh translations for new validation messages...

File: service-front/app/languages/cy/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -4978,3 +4988,108 @@ msgstr ""
 "Gallwch <a href=\"%view-lpa%\" class=\"govuk-link\">weld yr LPA hwn nawr</a> "
 "neu <a href=\"%add-lpa%\" class=\"govuk-link\">ychwanegu LPA arall</a> at eich "
 "cyfrif."
+
+#: src/Actor/templates/actor/partials/check-code-details.html.twig:10
+msgid "Access code created %date%"
+msgstr ""
```
Comment (GemTay) [inline service-front/app/languages/cy/LC_MESSAGES/messages.po:4994]: There is currently no welsh translation for this message - i will ask Uchenna for it and add it in a separate ticket when she gets it back from the translation unit

**Review by cooperaj — APPROVED** [review]

**Pull request merged**